Submitted By:            BLFS Book <blfs-book@linuxfromscratch.org>
Date:                    2004-09-21
Initial Package Version: 2.0.47
Upstream Status:         Not submitted (LFS specific)
Origin:                  BLFS Book
Description:             Modify the FHS layout for installing Apache

$LastChangedBy: randy $
$Date: 2004-10-18 22:34:20 -0600 (Mon, 18 Oct 2004) $

diff -Naur httpd-2.0.50-orig/config.layout httpd-2.0.50/config.layout
--- httpd-2.0.50-orig/config.layout	2004-07-15 15:07:48.000000000 +0000
+++ httpd-2.0.50/config.layout	2004-07-15 15:08:03.000000000 +0000
@@ -322,3 +322,26 @@
     installbuilddir: ${prefix}/etc/apache2/build
     errordir:      ${datadir}/error
 </Layout>
+
+<Layout FHS>
+    prefix:             /usr
+    exec_prefix:        ${prefix}
+    bindir:             ${exec_prefix}/bin
+    sbindir:            ${exec_prefix}/sbin
+    libdir:             ${exec_prefix}/lib
+    libexecdir:         ${exec_prefix}/lib/apache
+    mandir:             ${prefix}/share/man
+    sysconfdir:         /etc/apache
+    datadir:            /srv/www
+    installbuilddir:    ${libexecdir}/build
+    errordir:           ${datadir}/error
+    iconsdir:           ${datadir}/icons
+    htdocsdir:          ${datadir}/htdocs
+    manualdir:          ${datadir}/manual
+    cgidir:             ${datadir}/cgi-bin
+    includedir:         ${prefix}/include/apache
+    localstatedir:      ${datadir}
+    runtimedir:         /var/run
+    logfiledir:         /var/log/apache
+    proxycachedir:      /var/cache/apache/proxy
+</Layout>
diff -Naur httpd-2.0.50-orig/srclib/apr/config.layout httpd-2.0.50/srclib/apr/config.layout
--- httpd-2.0.50-orig/srclib/apr/config.layout	2004-07-15 15:07:48.000000000 +0000
+++ httpd-2.0.50/srclib/apr/config.layout	2004-07-15 15:08:03.000000000 +0000
@@ -11,18 +11,18 @@
 
 #   Classical APR path layout designed for parallel installs.
 <Layout apr>
-    prefix:        /usr/local/apr
+    prefix:        /usr
     exec_prefix:   ${prefix}
     bindir:        ${exec_prefix}/bin
-    sbindir:       ${exec_prefix}/bin
+    sbindir:       ${exec_prefix}/sbin
     libdir:        ${exec_prefix}/lib
-    libexecdir:    ${exec_prefix}/modules
-    mandir:        ${prefix}/man
-    sysconfdir:    ${prefix}/conf
-    datadir:       ${prefix}
-    installbuilddir: ${datadir}/build
-    includedir:    ${prefix}/include/apr-${APR_MAJOR_VERSION}
-    localstatedir: ${prefix}
+    libexecdir:    ${exec_prefix}/lib/apache
+    mandir:        ${prefix}/share/man
+    sysconfdir:    /etc/apache
+    datadir:       /srv/www
+    installbuilddir: ${libexecdir}/build
+    includedir:    ${prefix}/include/apache
+    localstatedir: ${datadir}
     libsuffix:     -${APR_MAJOR_VERSION}
 </Layout>
 
diff -Naur httpd-2.0.50-orig/srclib/apr-util/config.layout httpd-2.0.50/srclib/apr-util/config.layout
--- httpd-2.0.50-orig/srclib/apr-util/config.layout	2004-07-15 15:07:48.000000000 +0000
+++ httpd-2.0.50/srclib/apr-util/config.layout	2004-07-15 15:08:03.000000000 +0000
@@ -11,18 +11,18 @@
 
 #   Classical APR-util path layout designed for parallel installs.
 <Layout apr-util>
-    prefix:        /usr/local/apr
+    prefix:        /usr
     exec_prefix:   ${prefix}
     bindir:        ${exec_prefix}/bin
-    sbindir:       ${exec_prefix}/bin
+    sbindir:       ${exec_prefix}/sbin
     libdir:        ${exec_prefix}/lib
-    libexecdir:    ${exec_prefix}/modules
-    mandir:        ${prefix}/man
-    sysconfdir:    ${prefix}/conf
-    datadir:       ${prefix}
-    installbuilddir: ${datadir}/build
-    includedir:    ${prefix}/include/apr-${APRUTIL_MAJOR_VERSION}
-    localstatedir: ${prefix}
+    libexecdir:    ${exec_prefix}/lib/apache
+    mandir:        ${prefix}/share/man
+    sysconfdir:    /etc/apache
+    datadir:       /srv/www
+    installbuilddir: ${libexecdir}/build
+    includedir:    ${prefix}/include/apache
+    localstatedir: ${datadir}
     libsuffix:     -${APRUTIL_MAJOR_VERSION}
 </Layout>
 
